test_sai_sdk_dump fails after not finding an interface which is not a member of PC.

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/sonic-net/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**
When running on a testbed, this testcrashes if it cannot find an interface which is not a member of a portChannel.
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[show_techsupport/test_auto_techsupport.py:1273: in get_random_physical_port_non_po_member](https://dev.azure.com/mssonic/internal/_git/5380e8f7-6e2a-4154-8dee-f3be7b096894?path=show_techsupport%2Ftest_auto_techsupport.py&version=GBinternal&_a=contents&line=1273&lineEnd=1274&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
test_port = random.choice(non_po_ports)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = &lt;random.Random object at 0x5654d9850d40&gt;, seq = []
def choice(self, seq):
&quot;&quot;&quot;Choose a random element from a non-empty sequence.&quot;&quot;&quot;
&gt;       return seq[int(self.random() * len(seq))]  # raises IndexError if seq is empty
E       IndexError: list index out of range
self       = &lt;random.Random object at 0x5654d9850d40&gt;
seq        = []
<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**
1. run test_sai_sdk_dump test on a mellanox 4600 device where the configuration has all ports member of some or other portchannels.
2.
3.

**Describe the results you received:**
crash due to index out of range.

**Describe the results you expected:**
graceful failure with a message.

**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
    **Output of `show version`:**
 ```
SONiC Software Version: SONiC.internal.72539810-ee2e7de8f6
SONiC OS Version: 11
Distribution: Debian 11.7
Kernel: 5.10.0-18-2-amd64
Build commit: ee2e7de8f6
Build date: Sun Apr 30 17:09:43 UTC 2023
Built by: cloudtest@0645e225c000001

Platform: x86_64-mlnx_msn4600c-r0
HwSKU: ACS-MSN4600C
ASIC: mellanox
ASIC Count: 1
Serial Number: MT2115X26246
Model Number: MSN4600-CS2RO
Hardware Revision: A1
Uptime: 00:18:18 up 18:48,  1 user,  load average: 3.22, 2.83, 2.74
Date: Wed 03 May 2023 00:18:18
```
    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

--- a/tests/show_techsupport/test_auto_techsupport.py
+++ b/tests/show_techsupport/test_auto_techsupport.py
@@ -950,6 +950,7 @@ def get_new_techsupport_files_list(duthost, available_tech_support_files):
     :return: list of new techsupport files
     """
     try:
+        duthost.shell('ls -lh /var/dump/')  # print into logs full folder content(for debug purpose)
         new_available_tech_support_files = duthost.shell('ls /var/dump/*.tar.gz')['stdout_lines']
     except RunAnsibleModuleFail:
         new_available_tech_support_files = []
@@ -1233,6 +1234,10 @@ def clear_folders(duthost):
     Clear auto-techsupport related folders
     :param duthost: duthost object
     """
+    # print into logs folders content(for debug purpose) before remove
+    duthost.shell('sudo ls -lh /var/core/')
+    duthost.shell('sudo ls -lh /var/dump/')
+
     duthost.shell('sudo rm -rf /var/core/*')
     duthost.shell('sudo rm -rf /var/dump/*')
 
@@ -1266,11 +1271,13 @@ def get_random_physical_port_non_po_member(minigraph_facts):
     :return: string, port name
     """
     po_members = []
+    test_port = None
     for po_iface, po_data in list(minigraph_facts['minigraph_portchannels'].items()):
         po_members += po_data['members']
     all_ports = list(minigraph_facts['minigraph_ports'].keys())
     non_po_ports = [port for port in all_ports if port not in po_members]
-    test_port = random.choice(non_po_ports)
+    if non_po_ports:
+        test_port = random.choice(non_po_ports)
     return test_port
 
 

